python/libwholegraph/pyproject.toml

```
@@ -0,0 +1,90 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+[build-system]
+build-backend = "rapids_build_backend.build"
+requires = [
+    "rapids-build-backend>=0.3.0,<0.4.0.dev0",
+    "scikit-build-core[pyproject]>=0.10.0",
+] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
+
+[project]
+name = "libwholegraph"
+dynamic = ["version"]
+description = "wholegraph"
+readme = { file = "README.md", content-type = "text/markdown" }
+authors = [
+    { name = "NVIDIA Corporation" },
+]
+license = { text = "Apache-2.0" }
+requires-python = ">=3.10"
+classifiers = [
+    "Intended Audience :: Developers",
+    "Topic :: Database",
+    "Topic :: Scientific/Engineering",
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: C++",
+    "Environment :: GPU :: NVIDIA CUDA",
+]
+dependencies = [
+    "libraft==25.6.*,>=0.0.0a0",
+    "rapids-logger==0.1.*,>=0.0.0a0",
+] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
+
+
+[tool.setuptools.dynamic]
+version = {file = "libwholegraph/VERSION"}
+
+[project.urls]
+Homepage = "https://github.com/rapidsai/cugraph-gnn"
+
+[project.entry-points."cmake.prefix"]
+# this tells scikit-build-core to add this to cmake.prefix so
+# find_package(libwholegraph) can find the files from this wheel
+libwholegraph = "libwholegraph"
+
+[tool.pydistcheck]
+select = [
+    "distro-too-large-compressed",
+]
+
+# detect when package size grows significantly
+max_allowed_size_compressed = '1.2G'
```

```suggestion
max_allowed_size_compressed = '0.4G'
```

This size threshold should be reduced to more tightly match the current wheel sizes.

Looking through the `build-wheel-libwholegraph` logs ([build link](https://github.com/rapidsai/cugraph-gnn/actions/runs/14577429557/job/40886595329?pr=182)), I saw stuff like this for both CUDA 11 and CUDA 12 wheels:

```text
file size
  * compressed size: 0.3G
  * uncompressed size: 0.6G
  * compression space saving: 45.4%
```